packages/react-magma-dom/src/components/StyledButton/styles.ts

```diff
@@ -203,30 +255,44 @@ export function buildFocusBackground(props) {
 export function buildFocusColor(props) {
   props = buildPropsWithDefaultButtonStyles(props);
   if (props.color === 'marketing') {
+    if (props.variant !== 'solid' && props.isInverse) {
+      return props.theme.colors.tertiary500;
+    }
     return props.theme.colors.foundation02;
   }
+  if (props.color === 'danger') {
+    if (props.variant !== 'solid' && !props.isInverse) {
+      return darken(0.1, props.theme.colors.danger);
+    }
+  }
   if (
     (props.variant !== 'solid' && !props.isInverse) ||
     (props.variant === 'solid' && props.isInverse)
   ) {
     switch (props.color) {
       case 'secondary':
-        return props.theme.colors.neutral;
-      case 'success':
-        return darken(0.1, props.theme.colors.success);
+        return darken(0.1, props.theme.colors.primary);
+
       case 'danger':
-        return darken(0.1, props.theme.colors.danger);
+        return props.theme.colors.neutral100;
       default:
         return darken(0.1, props.theme.colors.primary);
     }
   }
   if (props.variant !== 'solid' && props.isInverse) {
-    return props.theme.colors.neutral08;
+    switch (props.color) {
+      case 'primary':
+        return props.theme.colors.tertiary500;
+      case 'secondary':
+        return props.theme.colors.tertiary500;
+      case 'danger':
+        return props.theme.colors.danger200;
+    }
```

Comment:
Move marketing check from line 257 to this switch statement, and make sure this switch has a default.

Reply:
Done